pkg/crawl/report.go

```diff
@@ -115,11 +116,10 @@ func (r *Reporter) Collect(ctx context.Context, path string, d fs.DirEntry, resu
 		}
 		// should not fail
 		jsonBytes, _ := json.Marshal(cveInfo)
-		output = string(jsonBytes)
+		_, _ = fmt.Fprintln(r.OutputWriter, string(jsonBytes))
 	} else {
-		output = fmt.Sprintf(cveMessage+" in file %s. log4j versions: %s. Reasons: %s", path, strings.Join(versions, ", "), strings.Join(readableReasons, ", "))
+		_, _ = fmt.Fprintln(r.OutputWriter, color.YellowString("[MATCH] "+cveMessage+" in file %s. log4j versions: %s. Reasons: %s", path, strings.Join(versions, ", "), strings.Join(readableReasons, ", ")))
```

Comment:
Similar to the above, this is using `%s` but in a `Fprintln` (rather than `Fprintf`), so think that `%s` will print literally.

Reply:
As above - I think the colour library is doing formatting and so we're getting the formatted strings printed out.